numpy/core/multiarray.py

```diff
@@ -945,15 +945,29 @@ def bincount(x, weights=None, minlength=None):
     >>> np.bincount(x).size == np.amax(x)+1
     True
 
-    The input array needs to be of integer dtype, otherwise a
-    TypeError is raised:
+    The input array needs have an integer dtype smaller or equal to the
+    native Python array index type, otherwise a TypeError is raised:
 
     >>> np.bincount(np.arange(5, dtype=float))
     Traceback (most recent call last):
       ...
     TypeError: Cannot cast array data from dtype('float64') to dtype('int64')
     according to the rule 'safe'
 
+    This also means that the following code will not work on a 32-bit system:
+
+    >>> import sys
+    >>> if sys.maxsize <= 2**32:
+    ...     np.bincount(np.arange(5, dtype=np.int64))
+    ...     # TypeError: Cannot cast array data from dtype('int64') to
+    ...     # dtype('int32') according to the rule 'safe'
+
+    The easiest way to solve this is by always using the Python int type
+    (or np.int_):
+
+    >>> np.bincount(np.arange(5, dtype=int))
+    array([1, 1, 1, 1, 1])
+
```

Comment:
```suggestion
```
I don't think this example is needed. Since the user might want to pass an int type other than the Python int type, this example can be confusing for them.

Reply:
You may be right, but I thought that the lack of a suitable example was even more confusing.
I can drop this example, though. The explicit mention of the int64 pitfall is probably enough.